src/rabbit_mgmt_event_collector_utils.erl

```diff
@@ -185,9 +185,12 @@ ceil(TS, #state{interval = Interval}) ->
 
 handle_created(TName, Stats, Funs) ->
     Formatted = rabbit_mgmt_format:format(Stats, Funs),
-    ets:insert(TName, {{id(TName, Stats), create},
-                                              Formatted,
-                                              pget(name, Stats)}).
+    Id = id(TName, Stats),
+    ets:insert(TName, {{Id, create}, Formatted, pget(name, Stats)}),
+    case lists:member(TName, ?PROC_STATS_TABLES) of
+        true  -> ets:insert(rabbit_mgmt_stats_tables:key_index(TName), {Id});
```

Comment:
Indexes are not strictly needed, as `channel_stats` and `connection_stats` only contain one entry per channel/connection. However, I see your point on using the index as an ordered_set, so you can traverse the table using first and next in the GC process. But the implementation needs to be amended, as I can't find where those indexes are deleted. The channel/connection_closed events have to remove the index entry in `handle_deleted`.